src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -0,0 +1,444 @@
+     *     New instance would have same values as {@code draft} except for supply channel reference. Reference
+     *     will be replaced with reference that points to ID of existing channel for key given in draft.</li>
+     *     <li>Empty if supply channel for key wasn't found in {@code supplyChannelKeyToId}</li>
+     * </ul>
+     *
+     * @param draft inventory entry draft from processed list
+     * @return {@link Optional} with draft that is prepared to being created or compared with existing
+     * {@link InventoryEntry}, or empty optional if method fails to find supply channel ID that should be referenced
+     */
+    private Optional<InventoryEntryDraft> replaceChannelReference(final InventoryEntryDraft draft) {
+        final String supplyChannelKey = SkuKeyTuple.of(draft).getKey();
+        if (supplyChannelKey != null) {
+            if (supplyChannelKeyToId.containsKey(supplyChannelKey)) {
+                return Optional.of(
+                    ofEntryDraftPlusRefToSupplyChannel(draft, supplyChannelKeyToId.get(supplyChannelKey)));
+            } else {
+                handleFailure(format("Failed to find supply channel of key '%s'", supplyChannelKey), null);
+                return Optional.empty();
+            }
+        }
+        return Optional.of(draft);
+    }
+
+    /**
+     * Returns new {@link InventoryEntryDraft} containing same data as {@code draft} except for
+     * supply channel reference that is replaced by reference pointing to {@code supplyChannelId}.
+     *
+     * @param draft           draft where reference should be replaced
+     * @param supplyChannelId ID of supply channel existing in target project
+     * @return {@link InventoryEntryDraft} with supply channel reference pointing to {@code supplyChannelId}
+     *      and other data same as in {@code draft}
+     */
+    private InventoryEntryDraft ofEntryDraftPlusRefToSupplyChannel(@Nonnull final InventoryEntryDraft draft,
+                                                                   @Nonnull final String supplyChannelId) {
+        final Reference<Channel> supplyChannelRef = Channel.referenceOfId(supplyChannelId);
+        return InventoryEntryDraftBuilder.of(draft)
+            .supplyChannel(supplyChannelRef)
+            .build();
+    }
+
+    /**
+     * Method tries to create supply channel of given {@code supplyChannelKey} in CTP project.
+     * If operation succeed then {@code supplyChannelKeyToId} map is updated, otherwise
+     * error callback function is executed.
+     *
+     * @param supplyChannelKey key of supply channel that seems to not exists in a system
+     * @return {@link CompletionStage} instance that indicates method progress
+     */
+    private CompletionStage<Void> createMissingSupplyChannel(@Nonnull final String supplyChannelKey) {
+        return inventoryService.createSupplyChannel(supplyChannelKey)
```

Comment:
here you're also modifying global variable. You're designing method to have side effects
(modifying those global vars)
instead of implementing in the way that it clearly produces some results